plugins/ubiquiti/unifi_api

```diff
@@ -0,0 +1,1138 @@
+	'enable_detail_clients_type'    => env_default_bool_true('enable_detail_clients_network'),
+	'hide_empty_xfer_port'          => env_default_bool_true('hide_empty_xfer_port'),
+	'show_authorized_clients_type'  => env_default_bool_true('show_authorized_clients_type'),
+);
+
+
+# Set up needed API endpoints
+my %APIPoint = (
+	'login'  => $APIconfig{"api_url"} . "/api/login",
+	'device' => $APIconfig{"api_url"} . "/api/s/" . $APIconfig{"site"} . "/stat/device",
+	'wlan'   => $APIconfig{"api_url"} . "/api/s/" . $APIconfig{"site"} . "/rest/wlanconf",
+	'sta'    => $APIconfig{"api_url"} . "/api/s/" . $APIconfig{"site"} . "/stat/sta",
+);
+
+my %APIResponse;
+my %APIJsonResponse;
+my %Data;
+my $retcode;
+
+# Init curl and JSON
+my $curl = WWW::Curl::Easy->new() or die "FATAL:$me: WWW::Curl::Easy init failed!\n";
+my $jsonOBJ = JSON->new() or die "FATAL:$me: JSON init failed!\n";
+
+
+## Fetch the data from the API
+
+# The rest is a way to use local files from the local disk. Undocumented and not really supported.
+
+if ( !env_default_bool_true('USE_API') ) {
+	if (! eval {require File::Slurp; File::Slurp->import(); 1; } ) {
+		die "Local debug unavailable, File::Slurp CPAN moduel required\n";
+	}
+	foreach ( "./demo-test-files/device", "./demo-test-files/sta", "./demo-test-files/wlanconf" ) {
+		if ( ! -f $_ ) { die "File not found: $_\n"; }
+	}
+	$APIJsonResponse{'device'} = $jsonOBJ->allow_nonref->utf8->relaxed->decode(read_file('./demo-test-files/device'));
+	$APIJsonResponse{'sta'} = $jsonOBJ->allow_nonref->utf8->relaxed->decode(read_file('./demo-test-files/sta'));
+	$APIJsonResponse{'wlan'} = $jsonOBJ->allow_nonref->utf8->relaxed->decode(read_file('./demo-test-files/wlanconf'));
+} else {
+	fetch_data();
+}
+
+
+## Process the data
+
+make_data();
+
+if ( defined($ARGV[0]) && $ARGV[0] eq "config" ) {
+	# Do the config step for each set of graphs
+	do_config_mem();
+	do_config_cpu();
+	do_config_load();
+	do_config_uptime();
+	do_config_xfer_by_device();
+	do_config_xfer_by_uplink();
+	do_config_xfer_by_port();
+	do_config_xfer_by_network();
+	do_config_xfer_by_radio();
+	do_config_clients_by_device();
+	do_config_clients_by_type();
+
+	# If dirtyconfig is not supported, or turned off, exit here.  Otherwise, continue to fetch section
+	if ( !defined($ENV{'MUNIN_CAP_DIRTYCONFIG'}) || !$ENV{'MUNIN_CAP_DIRTYCONFIG'} ) {  exit 0; }
+}
+
+# Do the fetch step for each set of graphs
+do_values_cpu();
+do_values_mem();
+do_values_load();
+do_values_uptime();
+do_values_xfer_by_device();
+do_values_xfer_by_uplink();
+do_values_xfer_by_port();
+do_values_xfer_by_network();
+do_values_xfer_by_radio();
+do_values_clients_by_device();
+do_values_clients_by_type();
+
+
+
+
+
+
+
+#######################
+# SUBROUTINES  CONFIG #
+#######################
+
+sub do_config_clients_by_type {
+	# Provide client count by type - CONFIG
+	if ( !$PluginConfig{'enable_clients_type'} ) { return 0; }
+
+	graph_prolouge(
```

Comment:
I think, it should be `prologue` instead.

Reply:
Hmm.  indeed.